install/helm/agones/templates/service/allocation.yaml

```diff
@@ -57,6 +57,10 @@ spec:
       heritage: {{ .Release.Service }}
   template:
     metadata:
+      annotations:
+{{- if .Values.agones.allocator.generateTLS }}
+        revision/tls-cert: {{ .Release.Revision | quote }}
+{{- end }}
```

Comment:
Actually can you explain how this annotation is causing restart?
Should annotations be inside the if statement?

Reply:
So this puts an annotation on each pod of the deployment - with the Helm revision number.

When you update helm with a new install, the revision number increases. Since there is now a new annotation on each underlying pod in the Deployment, kubernetes will do a rolling update to the new deployment, thus restarting everything.

`annotations` could also be in the if statement. Should work either way (it'll be an empty annotation block otherwise). I don't mind to be honest - if you feel like it should include the `annotations` block, I'll change it :+1:

Reply:
Thanks for the explanation.